--- gtk+3.0-3.24.5-orig/gtk/gtkcssshorthandpropertyimpl.c	2019-12-08 18:06:58.934357281 +0000
+++ gtk+3.0-3.24.5-new/gtk/gtkcssshorthandpropertyimpl.c	2019-12-08 17:30:57.648750023 +0000
@@ -1118,7 +1118,12 @@
       g_value_init (&v, G_TYPE_DOUBLE);
       size = pango_font_description_get_size (description) / PANGO_SCALE;
       if (!pango_font_description_get_size_is_absolute (description))
-        size = size * gdk_screen_get_resolution (gdk_screen_get_default ()) / 72.0;
+	{
+	  double dpi = gdk_screen_get_resolution (gdk_screen_get_default ());
+	  if (dpi <= 0.0)
+	    dpi = 96.0;
+	  size = size * dpi / 72.0;
+	}
       g_value_set_double (&v, size);
       prop = _gtk_style_property_lookup ("font-size");
       _gtk_style_property_assign (prop, props, state, &v);
